tests/Credentials/InstanceProfileProviderTest.php

```diff
@@ -1204,4 +1205,131 @@ public function testResetsAttempts()
         $provider()->wait();
         $this->assertLessThanOrEqual(3, $this->getPropertyValue($provider, 'attempts'));
     }
+
+    /**
+     * This test checks for disabling IMDSv1 fallback by explicit client config passing.
+     * @return void
+     */
+    public function testIMDSv1DisabledByExplicitConfig() {
+        $config = [InstanceProfileProvider::CFG_EC2_METADATA_V1_DISABLED => true];
+        $wereCredentialsFetched = $this->fetchMockedCredentialsAndAlwaysExpectAToken($config);
+
+        $this->assertTrue($wereCredentialsFetched);
+    }
+
+    /**
+     * This test checks for disabling IMDSv1 fallback by setting AWS_EC2_METADATA_V1_DISABLED to true.
+     * @return void
+     */
+    public function testIMDSv1DisabledByEnvironment() {
+        $ec2MetadataV1Disabled = ConfigurationResolver::env(InstanceProfileProvider::CFG_EC2_METADATA_V1_DISABLED, 'string');
+        $putIMDSv1VarValue = function ($value) {
+            putenv('AWS_' . strtoupper(InstanceProfileProvider::CFG_EC2_METADATA_V1_DISABLED) . '=' . $value);
+        };
+        $putIMDSv1VarValue('true');
+        try {
+            $wereCredentialsFetched = $this->fetchMockedCredentialsAndAlwaysExpectAToken();
+            $this->assertTrue($wereCredentialsFetched);
+        } finally {
+            $putIMDSv1VarValue($ec2MetadataV1Disabled);
+        }
+    }
+
+    /**
+     * This test checks for disabling IMDSv1 fallback by looking into the config file
+     * for the property aws_ec2_metadata_v1_disabled expected set to true.
+     * @return void
+     */
+    public function testIMDSv1DisabledByConfigFile() {
+        $currentConfigFile = getenv(ConfigurationResolver::ENV_CONFIG_FILE);
+        $mockConfigFile = "./mock-config";
+        try {
+            putenv(ConfigurationResolver::ENV_CONFIG_FILE . '=' . $mockConfigFile);
+            $configContent = "[default]" . "\n" . InstanceProfileProvider::CFG_EC2_METADATA_V1_DISABLED . "=" . "true";
+            file_put_contents($mockConfigFile, $configContent);
+            $wereCredentialsFetched = $this->fetchMockedCredentialsAndAlwaysExpectAToken();
+            $this->assertTrue($wereCredentialsFetched);
```

Comment:
I think we should just do away with try/finally blocks in both cases here unless we're going to handle exceptions.

Reply:
Actually the reason why I have a try-finally clause there is because I want to make sure that regardless what exception is thrown the finally block is executed. I know that in fetchMockedCredentialsAndAlwaysExpectAToken all exceptions thrown are caught there, and false is returned in such case, but I like to be safe. If is still required to remove it I can do it. 
Regarding the test passing without trying the assertion this is actually not possible, because an exception not handled will automatically make the test to fail, and also not having an assertion in the test is a failed case by default.

What do you folks think?

Reply:
I'm actually good with keeping that.  I see we do it in plenty of other places in our tests.  I don't think the file_exists check is necessary though.

Reply:
Perfect! 
I will remove the file_exists check.

Thanks!